opensearch-project/OpenSearch-Dashboards
# Add sidecar service

### Description

Support a new layout service called sidecar in dashboard, so that external plugin can use this service as its container. This flyout supports three modes and resizable.

### Issues Resolved

https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5620

## Screenshot

![image](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/42465692/3bb7ee1d-e80c-4ced-9948-ea3eb3b42462)
![image](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/42465692/63c3ae76-9865-43e4-8c66-ea25c8bfcb95)
![image](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/42465692/ebd5a816-aec4-4f8d-a552-cb3eabc9b072)


## Testing the changes



### Check List


### Test steps
1. Open sidecar: You could open sidecar in plugin or other application through `core.overlays.sidecar().open`, in this method you need to pass MountPoint and sidecar config, like
 `{
          className: 'chatbot-sidecar',
          config: {
            dockedMode: 'right',
            paddingSize: 460
    }`.
After that you could see a sidecar container is mounted.
2. You could drag border button to view resizable effect.
3. You could change docked Mode or docked padding size through calling `setSidecarConfig` with new config, like
`
  {
      dockedMode: 'takeover',
      paddingSize: 500
   }
`
4. You could call `hide` and `show` to implement sidecar hide and show effect.

Here is a test env. http://k8s-opensear-opensear-b09791709c-480119112.us-west-1.elb.amazonaws.com/app/home#/,  username: `admin`, password: `5fIavN2oFmWiy5E`
